### ISSUE TITLE
Gibbon::MailChimpError not populated with Faraday 0.16.2

### ISSUE THREAD

**Issue by millsaj** [amro/gibbon#283]
After updating Faraday to 0.16.2 (from 0.15.4) I ran into an issue where `Gibbon::MailChimpError` is not populated correctly. (I am using gibbon 3.2.0)

```ruby
#<Gibbon::MailChimpError: the server responded with status 404 @title=nil, @detail=nil, @body=nil, @raw_body=nil, @status_code=nil>
```

I found this whilst simulating 404 and 500 errors in our test suite using webmock as below:

```ruby
stub_request(:any, /mailchimp.com/)
  .and_return(status: 500, body: { status: 500 }.to_json)
```

**Comment by amro** [comment]
Hi Josh. Thanks. Sorry, but I didn’t have time to get to this today. Will try to find time to investigate tomorrow.

**Comment by drn** [comment]
@amro my team ran into this as well. I put up a quick PR to resolve

**Comment by amro** [comment]
Released version 3.3.0. Thank you for the bug report @millsaj and thank you for the contribution @drn !
